tools/downloader/topologies.py

```diff
@@ -0,0 +1,103 @@
+_cache_dir = Path(os.getenv('OPEN_MODEL_ZOO_CACHE_DIR', _download_dir / '.cache'))
+
+def load_model(model_name, precision=None):
+    '''
+    Downloads target model to the folder specified in _download_dir variable.
+    If there is already downloaded model - retrieves the model from cache.
+
+    Parameters
+    ----------
+        model_name
+            Target model name.
+        precision
+            Target model precision. By default is None.
+
+    Returns
+    -------    
+    An object of the Topology class which stores information about the downloaded model.
+    '''
+    parser = argparse.ArgumentParser()
+    args = Namespace(all=False, list=None, name=model_name, print_all=False)   
+    topology = common.load_models_from_args(parser, args)[0]
+
+    model_dir = _download_dir / topology.subdirectory
+
+    if not precision:
+        downloader.main(['--name=' + model_name,
+                        '--output_dir=' + str(_download_dir).replace('\\', '\\\\'), 
+                        '--cache_dir=' + str(_cache_dir).replace('\\', '\\\\')])
+        
+    return Topology(model_name, model_dir, precision, topology.precisions,
+                    topology.description, topology.license_url, topology.task_type)
+
+class Topology:     
+    '''
+    {}
+    Model task type: {}
+
+    License: {}
+    '''
+
+    def __init__(self, name, model_dir, precision, precisions, description, license_url, task_type):
+        self.name = name
+        self.model_dir = model_dir
+        self.precisions = precisions
+        self.precision = precision
+        Topology.__doc__ = Topology.__doc__.format(description, task_type, license_url)
+
+    def _get_ir(model_dir, precision, name):
+        prefix = Path(model_dir) / precision / name
+
+        xml_path = str(prefix) + '.xml'
+        bin_path = str(prefix) + '.bin'
+        if os.path.exists(xml_path) and os.path.exists(bin_path):
+            return xml_path, bin_path
+
+        converter.main(['--name=' + self.name, '--precisions=' + self.precision,
+                        '--download_dir=' + str(_download_dir).replace('\\', '\\\\')])
+
+        return xml_path, bin_path
+
+    def get_ir(self, precision=None):
+        '''
+        Creates OpenVINO Intermediate Representation (IR) network model.
+        If there is already converted model or the model has been downloaded in
+        IR - just returns the paths to the files.
+
+        Parameters
+        ----------
+            precision
+                Target model precision. By default is None.
+
+        Returns
+        -------
+        If precision was setted in this method or not None precision was setted in __init__:
+            xml_path
+                Path to generated or downloaded .xml file
+            bin_path
+                Path to generated or downloaded .bin file
+
+        Or returns list of pairs: (xml_path, bin_path) for all precisions of target model.
```

Comment:
`get_ir` should return single model. If no precision specified for `get_ir` we should return rather precision specified during  downloading or if it also None - default Topology's precision